lib/internal/process/promises.js

```diff
@@ -139,17 +139,19 @@ function emitDeprecationWarning() {
   }
 }
 
-// If this method returns true, at least one more tick need to be
-// scheduled to process any potential pending rejections
+// If this method returns true, we've executed user code or triggered
+// a warning to be emitted which requires the microtask and next tick
+// queues to be drained again.
 function processPromiseRejections() {
+  let maybeScheduledTicksOrMicrotasks = asyncHandledRejections.length > 0;
```

Comment:
On a side note, I wonder if it make sense now to just merge `process/promises.js` and `process/task_queues.js` - the latter is currently the only one who requires the former.

Reply:
Maybe? At the same time it's a little easier to read the task queue code as is. Also, the code in promises isn't really explicitly tied to task queues.

If someone feels strongly, they should open a PR. I, personally, like it as is.

Reply:
Keep in mind, more files = more startup time.

Reply:
@Fishrock123 This is now somewhat negligible since we have shipped pre-built code cache for builtin modules, which eliminates the majority of the overhead of loading an builtin module.